### PR TITLE
Fix gaussian integral

### PR DESCRIPTION
**Pull request by RainerHeintzmann**
fixed a bug with the normalization of Gaussian filtering. Added some tests, which also test the previously buggy case of zero-integral arrays.

### CODE REVIEW

File: src/fourier_filtering.jl [hunk 1]
```diff
@@ -90,7 +90,7 @@ function fourier_filter(arr, fct=window_gaussian;  kwargs...)
     return fourier_filter!(copy(arr), fct; kwargs...)
 end
 
-function fourier_filter_by_1D_FT!(arr::TA, wins::AbstractVector; transform_win=false, dims=(1:ndims(arr))) where {N, TA <: AbstractArray{<:Complex, N}}
+function fourier_filter_by_1D_FT!(arr::TA, wins::AbstractVector; transform_win=false, keep_integral=false, dims=(1:ndims(arr))) where {N, TA <: AbstractArray{<:Complex, N}}
```
Comment (roflmaostc) [inline src/fourier_filtering.jl]: Keep_integral?
Sounds a bit weird. Maybe something regarding normalization.

And, docstrings are missing.

Comment (RainerHeintzmann) [reply src/fourier_filtering.jl]: Is normalize_kernel better?

Comment (roflmaostc) [reply src/fourier_filtering.jl]: Yes, since we also work with discrete sums and not integrals here.

Comment (RainerHeintzmann) [reply src/fourier_filtering.jl]: to be consistent with the other arg, I now used "normalize_win".

Comment (RainerHeintzmann) [reply src/fourier_filtering.jl]: ... also added docstrings, but since these concerns only non-exported functions, I did not add them to the docs.

**Comment by sebaseb98** [comment]
Thank you for the fast implementation! Looks good to me :)

File: src/fourier_filtering.jl [hunk 2]
```diff
@@ -90,7 +90,19 @@ function fourier_filter(arr, fct=window_gaussian;  kwargs...)
     return fourier_filter!(copy(arr), fct; kwargs...)
 end
 
-function fourier_filter_by_1D_FT!(arr::TA, wins::AbstractVector; transform_win=false, dims=(1:ndims(arr))) where {N, TA <: AbstractArray{<:Complex, N}}
+"""
+    fourier_filter_by_1D_FT!(arr::TA, wins::AbstractVector; transform_win=false, normalize_win=false, dims=(1:ndims(arr))) where {N, TA <: AbstractArray{<:Complex, N}}
```
Comment (roflmaostc) [inline src/fourier_filtering.jl:94]: maybe `transform_win` and `normalize_win`.
I think being more verbose would be better here since `win` is confusing

File: src/fourier_filtering.jl [hunk 3]
```diff
@@ -100,17 +112,42 @@ function fourier_filter_by_1D_FT!(arr::TA, wins::AbstractVector; transform_win=f
...
+                    if (tmp[1] != 0 && tmp[1] != 1)
+                        tmp ./= tmp[1]
+                    end
+                end    
+                tmp
             else
-                wins[d]
+                if (normalize_win)
+                    if (wins[d][1] != 0 && wins[d][1] != 1)
+                        wins[d] ./ wins[d][1]
```
Comment (roflmaostc) [inline src/fourier_filtering.jl:125]: this line is not covered by tests

File: src/fourier_filtering.jl [hunk 4]
```diff
@@ -124,10 +161,22 @@ function fourier_filter_by_1D_FT!(arr::TA, fct=window_gaussian; dims=(1:ndims(ar
...
+filters an array by sequential multiplication in Fourierspace using one-dimensional RFFTs (first transform dimension) and FFTS (other transform dimensions) with one-dimensional kernels as specified in `wins`.
+
+#Arguments
++`arr`:     the array to filter by separable multiplication with windows.
++`wins`:    the window as a vector of vectors. Each of the original array dimensions corresponds to an entry in the outer vector.
++`transform_win`: specifies whether the directional windows each need to be FFTed before applying.
++`normalize_win`: specifies whether the directional windows (after potential FFT) each need to be normalized to a value of 1 at the zero frequency coordinate.
++`dims`: the dimensions to apply this separable multiplication to. If empty, the array will not be filtered.
+"""
+function fourier_filter_by_1D_RFT!(arr::TA, wins::AbstractVector; dims=(1:ndims(arr)), transform_win=false, normalize_win=false, kwargs...) where {T<:Real, N, TA<:AbstractArray{T, N}}
```
Comment (roflmaostc) [inline src/fourier_filtering.jl:179]: `windows` instead of `wins`

File: src/fourier_filtering.jl (hunk 2 above)
Comment (RainerHeintzmann) [reply src/fourier_filtering.jl:94]: Sorry. Don't understand your comment. It is called "normalize_win" and you suggest the same?
We should definitely not change transform_win to stay backwards compatible. So I agree that "win" should be more verbose but consistency is also important.

Comment (roflmaostc) [reply src/fourier_filtering.jl:94]: Ok didn't see this was already in the old version
